src/libs/actions/Session.js

```diff
@@ -218,7 +234,10 @@ function setPassword(password, validateCode) {
             }
 
             // This request can fail if the password is not complex enough
-            Onyx.merge(ONYXKEYS.SESSION, {error: response.message});
+            Onyx.merge(ONYXKEYS.ACCOUNT, {error: response.message});
+        })
+        .finally(() => {
+            Onyx.merge(ONYXKEYS.ACCOUNT, {loading: false});
```

Comment:
Just thinking out loud and not requesting a change. But seeing all of this logic makes me wonder if it might be valuable to create a higher order function or system that automatically applies this loading state logic to any API command by passing a related Onyx key. But not sure if we need that yet.